File: csharp/LogicAndTrick.WikiCodeParser/Nodes/PlainTextNode.cs
using System;
using System.Collections.Generic;

namespace LogicAndTrick.WikiCodeParser.Nodes
{
    /// <summary>
    /// Plain text. Contains only text and no other markup.
    /// </summary>
    public class PlainTextNode : INode
    {
        public static INode Empty => new PlainTextNode(string.Empty);

        public string Text { get; set; }

        public PlainTextNode(string text)
        {
            Text = text;
        }

        public string ToHtml() => HtmlHelper.Encode(Text);
        public string ToPlainText() => Text;
        public IList<INode> GetChildren() => Array.Empty<INode>();
        public void ReplaceChild(int i, INode node) => throw new InvalidOperationException();
        public bool HasContent() => !String.IsNullOrWhiteSpace(Text);
    }
}

File: php/src/Nodes/PlainTextNode.php
<?php

namespace LogicAndTrick\WikiCodeParser\Nodes;

use LogicAndTrick\WikiCodeParser\HtmlHelper;

class PlainTextNode implements INode
{
    public static function Empty(): INode
    {
        return new PlainTextNode('');
    }

    public static function NewLine(): INode
    {
        return new PlainTextNode("\n");
    }

    public string $text;

    public function __construct(string $text)
    {
        $this->text = $text;
    }

    function ToHtml(): string
    {
        return HtmlHelper::Encode($this->text);
    }

    function ToPlainText(): string
    {
        return $this->text;
    }

    function GetChildren(): array
    {
        return [];
    }

    function ReplaceChild(int $i, INode $node): void
    {
        throw new \Exception('Invalid operation');
    }

    function HasContent(): bool
    {
        return $this->text && trim($this->text) != '';
    }
}